src/modules/gnd_pos_control/gnd_pos_control_main.cpp

```diff
@@ -0,0 +1,1405 @@
+
+	/**
+	 * Destructor, also kills the sensors task.
+	 */
+	~GroundRoverPositionControl();
+
+	// prevent copying
+	GroundRoverPositionControl(const GroundRoverPositionControl &) = delete;
+	GroundRoverPositionControl operator=(const GroundRoverPositionControl &other) = delete;
+
+	/**
+	 * Start the sensors task.
+	 *
+	 * @return	OK on success.
+	 */
+	static int	start();
+
+	/**
+	 * Task status
+	 *
+	 * @return	true if the mainloop is running
+	 */
+	bool		task_running() { return _task_running; }
+
+private:
+	orb_advert_t	_mavlink_log_pub;
+
+	bool		_task_should_exit;		/**< if true, sensor task should exit */
+	bool		_task_running;			/**< if true, task is running in its mainloop */
+
+	int		_global_pos_sub;
+	int		_pos_sp_triplet_sub;
+	int		_ctrl_state_sub;			/**< control state subscription */
+	int		_control_mode_sub;		/**< control mode subscription */
+	int		_vehicle_command_sub;		/**< vehicle command subscription */
+	int		_vehicle_status_sub;		/**< vehicle status subscription */
+	int		_vehicle_land_detected_sub;	/**< vehicle land detected subscription */
+	int		_params_sub;			/**< notification of parameter updates */
+	int		_manual_control_sub;		/**< notification of manual control updates */
+
+	orb_advert_t	_attitude_sp_pub;		/**< attitude setpoint */
+	orb_advert_t	_tecs_status_pub;		/**< TECS status publication */
+	orb_advert_t	_gnd_pos_ctrl_status_pub;		/**< navigation capabilities publication */
+
+	orb_id_t _attitude_setpoint_id;
+
+	struct control_state_s				_ctrl_state;			/**< control state */
+	struct vehicle_attitude_setpoint_s		_att_sp;			/**< vehicle attitude setpoint */
+	struct fw_pos_ctrl_status_s			_gnd_pos_ctrl_status;		/**< navigation capabilities */
+	struct manual_control_setpoint_s		_manual;			/**< r/c channel data */
+	struct vehicle_control_mode_s			_control_mode;			/**< control mode */
+	struct vehicle_command_s			_vehicle_command;		/**< vehicle commands */
+	struct vehicle_status_s				_vehicle_status;		/**< vehicle status */
+	struct vehicle_land_detected_s			_vehicle_land_detected;		/**< vehicle land detected */
+	struct vehicle_global_position_s		_global_pos;			/**< global vehicle position */
+	struct position_setpoint_triplet_s		_pos_sp_triplet;		/**< triplet of mission items */
+
+	perf_counter_t	_loop_perf;			/**< loop performance counter */
+
+	float	_hold_alt;				/**< hold altitude for altitude mode */
+	float	_hdg_hold_yaw;				/**< hold heading for velocity mode */
+	bool	_hdg_hold_enabled;			/**< heading hold enabled */
+	bool	_yaw_lock_engaged;			/**< yaw is locked for heading hold */
+	float	_althold_epv;				/**< the position estimate accuracy when engaging alt hold */
+	struct position_setpoint_s _hdg_hold_prev_wp;	/**< position where heading hold started */
+	struct position_setpoint_s _hdg_hold_curr_wp;	/**< position to which heading hold flies */
+	hrt_abstime _control_position_last_called; 	/**<last call of control_position  */
+
+	/* Takeoff launch detection and runway */
```

Comment:
maybe the launch logic may be applied but without considering a "takeoff" (since it is a UGV)